source-archive: Correct version modification time and short version

rleigh-dundee opened the pull request:
Correct newline stripping and timestamp

```
scripts/source-archive --release=test --target=/tmp --tag=HEAD
```

Check `cmake/GitVersion.cmake` in the tar archive.  The shortversion will now not have a trailing newline.  The version will also be the current date and time, rather than 1-1-1970.  While that shouldn't have been problematic, it was breaking the Ninja build.

sbesson commented on scripts/source-archive:
```diff
@@ -186,6 +186,7 @@ if __name__ == "__main__":
         vcsdate_unix, vcsdate))
     cmakeversion = tarfile.TarInfo("%s/cmake/GitVersion.cmake" % (prefix))
     cmakeversion.size = cmakeversionbuf.len
+    cmakeversion.mtime = time.time()
```
Can you add a comment in the script itself explaining the reason why the `mtime` attribute of `TarInfo` need to be overriden?

sbesson commented:
Is the file removal required in this context or could we have the cleanup done as a separate PR?

Commit:
source-archive: Set GitVersion mtime

The Ninja build tool can't cope with an zero mtime, so
ensure it's set to the current time when repacking the
archive (see
https://github.com/ninja-build/ninja/issues/1120).

ghost commented:
@sbesson Yikes, no.  That's unrelated for another card.  I've set it back to the minimal change which was intended.

ghost commented:
For reference: https://github.com/ninja-build/ninja/issues/1120 this is the bug report for the Ninja behaviour.

sbesson commented:
Retested with the last version of the commit. Works as expected.
